src/audio/buffer.c

```diff
@@ -195,10 +195,22 @@ void buffer_free(struct comp_buffer *buffer)
 	rfree(buffer);
 }
 
+/*
+ * comp_update_buffer_produce() and comp_update_buffer_consume() send
+ * NOTIFIER_ID_BUFFER_PRODUCE and NOTIFIER_ID_BUFFER_CONSUME notifier events
+ * respectively. The only recipient of those notifications is probes. The
+ * target for those notifications is always the current core, therefore notifier
```

Comment:
The above is true, but whether it's actually what we want, is a bit complicated. And this is at heart  of #6100 .

Notifiers are per-cpu objects and currently the probe core call notify_register() only on one  core. This simply won't work if the buffer is attached to a pipeline running on a different core.

And once you do the registration correctly, then you have a whole new set of problems in probe.c code as it needs to multiple data from multiple cores into a single DMA stream, and do it without trashing performance. 

So this is probably the right change for current code, but when we implement multicore support #6100 , then we might need a revisit as the recipient needs to flush out the data so it can be passed to destionation core (that does the probe DMA to host). 

There are of course many alternatives. We could also replicate the DMA buffer for each core and have multiple cores drive the DMA with some locking mechanism.

Reply:
@kv2019i I think you are suggesting we need a ```buffer_acquire_readonly()``` which will take a snapshot of the current state and return a RO pointer. @lyakh I think this would need a new sparse RO type too.

Reply:
@lgirdwood not sure I understand. For RO you'd still need to lock, because otherwise you can get inconsistent data, and if you lock you're back to the dead-lock problem?